src/Service.Tests/SqlTests/RestApiTests/Put/PutApiTestBase.cs

```diff
@@ -175,7 +176,76 @@ public virtual async Task PutOneUpdateAccessibleRowWithDatabasePolicy()
                     sqlQuery: GetQuery("PutOneUpdateAccessibleRowWithDatabasePolicy"),
                     operationType: Config.Operation.Upsert,
                     requestBody: requestBody,
-                    clientRoleHeader: "policy_tester_REST"
+                    clientRoleHeader: "policy_tester_REST",
+                    expectedStatusCode: HttpStatusCode.OK
+                    );
+        }
+
+        /// <summary>
+        /// Test to validate that PUT operation which satisfies either or both the database policies for update/create operation
+        /// executes successfully.
+        /// </summary>
+        [TestMethod]
+        public virtual async Task PutOneWithDatabasePolicy()
+        {
+            // Validate that PUT operation satisfying the database policy for update operation ("@item.categoryid ne 0")
+            // and having a record corresponding to provided primary key successfully updates the record.
+            string requestBody = @"
+            {
+                ""categoryName"": ""SciFi"",
+                ""piecesRequired"": 5,
+                ""piecesAvailable"": 8
+            }";
+
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: "categoryid/100/pieceid/0",
+                    queryString: null,
+                    entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
+                    sqlQuery: GetQuery("PutOneUpdateWithUpdateDatabasePolicySatisfied"),
+                    operationType: Config.Operation.Upsert,
+                    requestBody: requestBody,
+                    clientRoleHeader: "database_policy_tester",
+                    expectedStatusCode: HttpStatusCode.OK
+                    );
+
+            // Validate that PUT operation on a table which does not satisfy database policy for update operation ("@item.categoryid ne 0")
```

Comment:
What if the record to be inserted is already present? (when the insert policy satisifies but update fails)

Reply:
In that case, we would throw an exception that we couldn't update or insert a record.